be/src/runtime/data_stream_recvr.cc

```diff
@@ -441,15 +351,13 @@ Status DataStreamRecvr::SenderQueue::_add_chunks_internal(const PTransmitChunkPa
         wait_timer.stop();
 
         for (auto& pair : chunks) {
-            _chunk_queue.emplace_back(std::move(pair));
+            ChunkItem item {pair.first, std::move(pair.second), nullptr};
+            _chunk_queue.emplace_back(std::move(item));
         }
-        // if done is nullptr, this function can't delay this response
         if (done != nullptr && _recvr->exceeds_limit(total_chunk_bytes)) {
```

Comment:
```suggestion
        if (!chunks.empty() && done != nullptr && _recvr->exceeds_limit(total_chunk_bytes)) {
```
We cannot assume that chunks are non-empty.

Reply:
Fixed